### DISCUSSION TITLE
How to change the historical weather data in Africa (Keyna) to the future weather data (from a climate model) in pypsa-earth?

### DISCUSSION THREAD

**Discussion by Haiping007** [pypsa-meets-earth/pypsa-earth#953] [Q&A]
Hi, guys, Happy New Year!

As we want to do power system modeling (for Kenya) integrated with a climate model, we need to use future weather data for this power system modeling.
Now, I get some preliminary data (Africa in 2050) from the climate model. So my first thinking is to update the 'cutout-2013-era5.nc', 'renewable_profiles', and the energy demand (ssp2-2.6) under the historical weather to the data of/under future weather.

I have no any experiences with this updating. I have a few questions:
1. The data of 'cutout-2013-era5.nc' and 'renewable_profiles' are only for the defined region (Kenya)? right?
2. Do you have any suggestions (such as how to process the updating?  how to keep the consistency of the data in formats, resolutions with netCDF4 file, or any other data/files that needs to be updated)? 

Thank you!

**Comment by ekatef** [comment]
Hey @Haiping007!

If you are using the supplied with the model, it covers the whole African continent. You can find it in `pypsa-earth/cutouts` folder and investigate it as a normal xarray file. You can read more on cutout generation in atlite docs: https://atlite.readthedocs.io/en/latest/examples/create_cutout.html 

Regarding incorporation climate projections, we are currently working on it. You are absolutely welcome to contribute! Happy to provide any details, if you are interested.

**Reply by Haiping007** [reply]
Thank you for being so supportive! I will check the docs @ekatef 
I might get in touch with you again once I get the climate data from my colleague.

**Reply by ekatef** [reply]
My pleasure 🙂

In the context of climate modeling, we are also considering more advanced climate-related features like implementation of bias correction approaches or creating options to account for multi-annual climate variability instead of taking a single year only. It would be great to understand if that could be helpful to some practical-relater tasks. That is precious to have opinions in that regard from people potentially interested in using and developing such features. Happy to continue anytime.

**Reply by Haiping007** [reply]
@ekatef Hi, my colleague is preparing the future climate data. We had a look at the cutout file and the renewable profile files downloaded in pypsa-earth. We have a few questions:
1. Our thought is that we create a future climate data file similar to the cutout file in terms of structure and variables, then code to adjust or recreate the renewable profile files, also adjust the demand based on the future weather. Finally, we run the model again?  Is this path right for modeling based on future weather profiles?
2. There is a variable called 'soil temperature' in the cutout file. We know it refers to 'Soil temperature between 1 meter and 3 meters depth (layer 4)'. But may I ask what the model uses the soil temperature variable for?
Thank you!

**Reply by davide-f** [reply]
Great @Haiping007 :D
Would you like to discuss on that at the end of our recurrent meetings? May you have time? The next one is on Thursday at 4pm UK time. If you are short on time, you could join about 30 minutes late.
It would be great to have your perspective and suggest on the way forward in a coordinated way :)

**Reply by Haiping007** [reply]
@davide-f  Yes, I can join in for about 30 minutes and leave early.
Is it tomorrow at 4pm UK time? Please send me a link.

**Reply by davide-f** [reply]
Fantastic!
The meeting is on discord: https://discord.com/invite/Br2xydE9

If you send me a pm there with your email, I can add you to the calendar meetings :)
My id on discord is davidsf

**Reply by Haiping007** [reply]
@davide-f  
I am very sorry. I have something urgent to deal with that I couldn't attend the meeting today. 
Could we arrange another time?  Thank you!

**Reply by davide-f** [reply]
Hello @Haiping007 !
No problem, we can definitely find another time.
What availabilities do you have in the coming days? feel free to reply me on discord

**Reply by Haiping007** [reply]
Hi, @davide-f  Thank you!
I am available at 10:00-12:00 and 14:00-15:00 on Monday and most time on Tuesday.
Any time slot fits you? (I also messaged you on discord)
